operator/controllers/seldondeployment_controller.go

```diff
@@ -554,6 +578,21 @@ func (r *SeldonDeploymentReconciler) createComponents(mlDep *machinelearningv1.S
 
 			}
 
+			// Find the current deployment and add the environment variables for the certificate
+			if len(certSecretRefName) > 0 {
+				sPodSpec, idx := utils.GetSeldonPodSpecForPredictiveUnit(&p, p.Graph.Name)
+				currentDeployName := machinelearningv1.GetDeploymentName(mlDep, p, sPodSpec, idx)
+				for i := 0; i < len(c.deployments); i++ {
+					d := c.deployments[i]
+					if strings.Compare(d.Name, currentDeployName) == 0 {
+						utils.MountSecretToDeploymentContainers(d, certSecretRefName, envDefaultCertMountPath)
+						certEnvVar := &corev1.EnvVar{Name: SELDON_MOUNT_PATH_ENV_NAME, Value: envDefaultCertMountPath}
+						utils.AddEnvVarToDeploymentContainers(d, certEnvVar)
+						break
+					}
+				}
+			}
```

Comment:
Do we also need to cover cases where separate parts of the same inference graph are split across multiple deployments? Would this be enough to cover all bases? Maybe @cliveseldon can help on this point.

Reply:
From the implementation workflow this should cover both cases - it would still be good to get @cliveseldon confirm that this is the case